### ISSUE TITLE
az login hangs

### ISSUE THREAD

**Issue by apsteger** [Azure/azure-cli#9852]
## Describe the bug

**Command Name**
`az login`

**Errors:**
Running 'az login' hangs. It does not exit with a timeout or error code.

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az login`
- browser opens
- click on user name and log in
- After selecting user, close browser or wait 10 seconds for redirect
- az login command still has not returned

## Expected Behavior
After clicking username in browser, it should perform a login and return control to the user in powershell environment

## Environment Summary
```
Windows-10-10.0.17134-SP0
Python 3.6.6
Shell: powershell.exe

azure-cli 2.0.63 *

```

**Comment by marstr** [comment]
Howdy @apsteger,

We've seen issues like this before when folks are using the CLI behind a VPN or proxy. Please checkout the following troubleshooting steps, and let me know if you're still encountering this issue: https://github.com/Azure/azure-cli/blob/dev/doc/use_cli_effectively.md#working-behind-a-proxy

**Comment by Nash-Vegas** [comment]
This was happening to us until we entered 
>export HTTPS_PROXY="proxy.server.com"
>export HTTP_PROXY="proxy.server.com"
at the CLI.

**Comment by tommysundling** [comment]
I'm also encountering an issue where "az login" hangs in Visual Studio Code (v1.36.0) when running in a Powershell context.

When I run "az login" it starts my default browser (Edge), I then select my user account (@hotmail) after which the browser just sits there with the loading animation. The VS Code window also never returns or timeouts from "running Powershell command".

If possible I would like to be able to use VS Code as my editor for my Azure CLI scripts but I'm not sure what is going wrong.

**Comment by apsteger** [comment]
Thanks @marstr , I'll give it a try.

**Comment by marstr** [comment]
I'm going to close this for now, but I'll be happy to re-open if more information comes to light.

@tommysundling, I'd also try messing around with the troubleshooting guide.

**Comment by barnstee** [comment]
I'm also facing this issue and it hangs wither AFTER I do my 2-factor auth and the login windows in the browser stays at the progress bar screen or it says you are login in and moves on to the Azure CLI homepage. In either case, there is no further output on the Powershell Windows, even with --debug enabled. I've made sure that I am logged out before attempting another login, too. The last line of output, even with --debug enabled, is:
Note, we have launched a browser for you to login. For old experience with device code, use "az login --use-device-code"
